### DISCUSSION TITLE
Internal Runtime Error when I try to execute a function

### DISCUSSION THREAD

**Discussion by LeonardoMantovani** [appwrite/appwrite#3925] [Q&A] [answered]
Hi, I'm using AppWrite 1.0 with the Python 3.9 runtime (deployed on a DigitalOcean VPS with Fedora36, 2 GB of RAM and 1CPU).

I'm opening this Q&A because when I try to execute a cloud function I wrote (that works flawlessly if I run it as a local python function on my computer) I get the `Internal Runtime Error` output on STDERR

I went checking the logs as suggested [in this thread](https://github.com/appwrite/appwrite/issues/2940) and this is what I found:
```
appwrite-executor  | Warming up Node.js 16.0 environment...
appwrite-executor  | Warming up PHP 8.0 environment...
appwrite-executor  | Warming up Ruby 3.0 environment...
appwrite-executor  | Warming up Python 3.9 environment...
appwrite-executor  | Warming up Dart 2.17 environment...
appwrite-executor  | Removing orphan runtimes...
appwrite-executor  | Successfully Warmed up Python 3.9!
appwrite-executor  | Successfully Warmed up Dart 2.17!
appwrite-executor  | Successfully Warmed up Ruby 3.0!
appwrite-executor  | Successfully Warmed up Node.js 16.0!
appwrite-executor  | Successfully Warmed up PHP 8.0!
appwrite-executor  | Successfully removed presents-632f27980becd2e21abf
appwrite-executor  | Successfully removed presents-632f2bae8c7e3b2ca944
appwrite-executor  | [Error] Type: Exception
appwrite-executor  | [Error] Message: Runtime not found. Please create the runtime.
appwrite-executor  | [Error] File: /usr/src/code/app/executor.php
appwrite-executor  | [Error] Line: 469
appwrite-executor  | Building container : presents-632f2bae8c7e3b2ca944
appwrite-executor  | Build Stage completed in 2 seconds
appwrite-executor  | Executing Runtime: presents-632f2bae8c7e3b2ca944
appwrite-executor  | [Error] Type: Exception
appwrite-executor  | [Error] Message: An internal curl error has occurred within the executor! Error Msg: Connection refused
appwrite-executor  | [Error] File: /usr/src/code/app/executor.php
appwrite-executor  | [Error] Line: 541
appwrite-executor  | [Error] Type: Exception
appwrite-executor  | [Error] Message: An internal curl error has occurred within the executor! Error Msg: Connection refused
appwrite-executor  | [Error] File: /usr/src/code/app/executor.php
appwrite-executor  | [Error] Line: 541
appwrite-executor  | Executing Runtime: presents-632f2bae8c7e3b2ca944
appwrite-executor  | Executing Runtime: presents-632f2bae8c7e3b2ca944
appwrite-executor  | Function executed in 0.022143125534058 seconds, status: failed
```
This is when I execute the function for the first time, then the next times the log output is just
```
appwrite-executor  | Executing Runtime: presents-632f2bae8c7e3b2ca944
appwrite-executor  | Function executed in 0.022143125534058 seconds, status: failed
```

I've read in the mentioned thread that it may be something wrong with the `appwrite_runtimes` network, but it seems up and connected to both the `appwrite-executor` container and the container created for the cloud function, so I have no idea what else should I look for... 

**any help?**

**Comment by stnguyen90** [comment, marked as answer]
@LeonardoMantovani can you provide the logs for your runtime container? It should be named `<project ID>-<deployment ID>`. Looking at your logs, it should be `presents-632f2bae8c7e3b2ca944`.

**Reply by LeonardoMantovani** [reply]
Thank you! I didn'r thought about looking at the logs of that container.
It seems python doesn't see the other file I uploaded together with `index.py` called `extra_functions.py` and throws a `ModuleNotFound` error when I try to `import extra_functions` at the beginning of `index.py`
it's weird because on my local PC that import worked just fine, but I guess I can move all functions in `index.py` to make it work on AW...

**Reply by stnguyen90** [reply]
@LeonardoMantovani, for Python, it would be best to use relative imports. For example, if `index.py` and `extra_functions.py` were in the same folder, I think doing `import .extra_functions` would work.
